test/integration_test/basic_test.go

```diff
@@ -16,9 +18,58 @@ import (
 )
 
 func TestStorageClusterBasic(t *testing.T) {
+	testutil.TestSpecPath = ""
 	t.Run("InstallWithAllDefaults", testInstallWithAllDefaults)
 	t.Run("NodeAffinityLabels", testNodeAffinityLabels)
 	t.Run("Upgrade", testUpgrade)
+	t.Run("InstallMetricsCollector", testInstallMetricsCollector)
+}
+
+func testInstallMetricsCollector(t *testing.T) {
+	secret := testutil.GetExpectedSecret(t, "/specs/pure-telemetry-cert.yaml")
+	require.NotNil(t, secret)
+
+	_, err := coreops.Instance().GetSecret(component.TelemetryCertName, pxNamespace)
+	require.True(t, err == nil || errors.IsNotFound(err))
+
+	if errors.IsNotFound(err) {
+		secret, err = coreops.Instance().CreateSecret(secret)
+		require.NoError(t, err)
+	} else {
+		secret, err = coreops.Instance().UpdateSecret(secret)
+		require.NoError(t, err)
+	}
```

Comment:
Why bother with a get and then create or update. Let's just create on start and delete on cleanup. The test should just assume the secret is not present when it starts.

Reply:
this is more robust, the create may fail if test exists for any reason and we rerun. I will add deletion in cleanup anyway.